src/Open3D/GUI/FileDialog.cpp

```diff
@@ -408,6 +408,7 @@ void FileDialog::SetOnDone(std::function<void(const char *)> onDone) {
 void FileDialog::OnDone() {
     if (this->impl_->onDone) {
         auto dir = this->impl_->CalcCurrentDirectory();
+        utility::filesystem::ChangeWorkingDirectory(dir);
```

Comment:
Well, it's the current *working* directory, and the user loaded a file from that directory, so that seems that the user chose to change the working directory. This is also the behavior of Windows (see https://devblogs.microsoft.com/oldnewthing/20101112-00/?p=12293), at least as of Windows XP, it may have changed. I don't know what macOS does. But I can see how it might be unexpected.